File: Code/Linear Regression/simple_linear_regression.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-

# Simple Linear Regression

# Importing the libraries
import numpy as np
import matplotlib.pyplot as plt
import pandas as pd

# Importing the dataset
dataset = pd.read_csv("breslow.csv")
X = dataset.iloc[:, 5].values
y = dataset.iloc[:, 1].values

X = X.reshape(-1, 1)
# Splitting the dataset into the Training set and the Test set
from sklearn.model_selection import train_test_split
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size = 0.25,
                                                    random_state = 0)

# Feature Scaling
"""from sklearn.preprocessing import StandardScaler
sc_X = StandardScaler()
X_train = sc_X.fit_transform(X_train)
X_test = sc_X.fit_transform(X_test)"""

# Fitting Simple Linear Regression to the Training set
from sklearn.linear_model import LinearRegression
regressor = LinearRegression()
regressor.fit(X_train, y_train)

# Predicting the Test set results
# Different from y_test
# y_test is the real data observed
# y_pred is the predicted data from the Linear Regression model
y_pred = regressor.predict(X_test)

# Visualising the Training set results
plt.scatter(X_train, y_train, color = 'red')
plt.plot(X_train, regressor.predict(X_train), color = 'blue')
plt.title('The number of smoker vs Age (Training Set)')
plt.xlabel('The number of smoker')
plt.ylabel('Age')
plt.show

# Visualising the Test set results
plt.scatter(X_test, y_test, color = 'red')
plt.plot(X_train, regressor.predict(X_train), color = 'blue')
plt.title('The number of smoker vs Age (Test Set)')
plt.xlabel('The number of smoker')
plt.ylabel('Age')
plt.show

File: Code/Regression/simple_linear_regression.r
  # Importing the data-set
  dataset = read.csv("/Users/Scylidose/Desktop/ML Folder/Linear Regression/breslow.csv")
  
  # Splitting the dataset into the Training set and the Test set
  # install.packages('caTools')
  library(caTools)
  set.seed(123)
  split = sample.split(dataset$age, SplitRatio = 2/3)
  training_set = subset(dataset, split == TRUE)
  test_set = subset(dataset, split == FALSE)
  
  # Feature Scaling
  # training_set[, 3:6] = scale(training_set[, 3:6])
  # test_set[, 3:6] = scale(test_set[, 3:6])
  
  # fitting Simple Linear Regression to the Training set
  regressor = lm(formula = age ~ ns, 
                 data = training_set)
  
  # Predicting the Test set results
  y_pred = predict(regressor, newdata = test_set)
  
  # Visualising the Training set results
  # install.packages('ggplot2')
  library(ggplot2)
  ggplot() + geom_point(aes(x = training_set$ns, y = training_set$age),
                        colour = 'red') + 
              geom_line(aes(x = training_set$ns, y = predict(regressor, newdata = test_set)),
                        colour = 'blue') + 
              ggtitle('The number of smoker vs Age (Training Set)') +
                xlab('The number of smoker') + 
                  ylab('Age')
  
  # Visualising the Test set results
  ggplot() + geom_point(aes(x = test_set$ns, y = test_set$age),
                        colour = 'red') + 
    geom_line(aes(x = training_set$ns, y = predict(regressor, newdata = test_set)),
              colour = 'blue') + 
    ggtitle('The number of smoker vs Age (Test Set)') +
    xlab('The number of smoker') + 
    ylab('Age')